pkg/controller/deployment/generic_deployment_controller.go

```diff
@@ -0,0 +1,98 @@
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/cert-manager-operator/pkg/operator/kubeclient"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/apiserver/controller/workload"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/klog/v2"
+)
+
+type genericDeploymentController struct {
+	kubeClientContainer kubeclient.KubeClientContainer
+	operatorClient      v1helpers.OperatorClient
+
+	assetFiles     []string
+	deploymentFile string
+}
+
+func newGenericDeploymentController(operatorNamespace, targetNamespace, targetVersion string, deploymentFile string, assetFiles []string, operatorClient v1helpers.OperatorClient, kubeClientContainer kubeclient.KubeClientContainer, kubeInformersForTargetNamespace informers.SharedInformerFactory, openshiftClusterConfigClient configv1.ClusterOperatorInterface, eventsRecorder events.Recorder, versionRecorder status.VersionGetter) factory.Controller {
+	controller := &genericDeploymentController{
+		kubeClientContainer: kubeClientContainer,
+		operatorClient:      operatorClient,
+
+		deploymentFile: deploymentFile,
+		assetFiles:     assetFiles,
+	}
+
+	return workload.NewController(
+		"cert-manager",
+		operatorNamespace,
+		targetNamespace,
+		targetVersion,
+		"",
+		"CertManager",
+		operatorClient,
+		kubeClientContainer.KubeConfig,
+		kubeInformersForTargetNamespace.Core().V1().Pods().Lister(),
+		[]factory.Informer{
+			operatorClient.Informer(),
+		},
+		[]factory.Informer{
+			kubeInformersForTargetNamespace.Apps().V1().Deployments().Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Namespaces().Informer(),
+		},
+		controller,
+		openshiftClusterConfigClient,
+		eventsRecorder,
+		versionRecorder,
+	)
+}
+
+func (c *genericDeploymentController) Sync(ctx context.Context, syncContext factory.SyncContext) (*appsv1.Deployment, bool, []error) {
+	klog.Infof("Triggering reconciliation loop")
+	var errors []error
+	var err error
+	var appliedDeployment *appsv1.Deployment
+
+	if len(c.assetFiles) > 0 {
```

Comment:
use this instead: https://github.com/openshift/library-go/blob/bec726e57e7414befca96d9ac3c156faff622bf8/pkg/operator/staticresourcecontroller/static_resource_controller.go#L68

Reply:
Done